src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/GetCoveragesValue.java

```diff
@@ -0,0 +1,230 @@
+                            min = 0,
+                            max = 1)
+                    String crs)
+            throws IOException {
+        CoverageInfo ci = null;
+        CoordinateReferenceSystem coordinateReferenceSystem = null;
+        String[] names = name.split(",");
+        if (names.length == 0) {
+            throw new WPSException("No coverage names provided");
+        }
+        ci = catalog.getCoverageByName(names[0]);
+        if (ci == null) {
+            throw new WPSException("Could not find coverage " + names[0]);
+        }
+        GridCoverageReader reader = ci.getGridCoverageReader(null, null);
+        GridCoverage2DReader reader2d = (GridCoverage2DReader) reader;
+        GridCoverage2D gridCoverage2D = null;
+        boolean nodata = false;
+        Number[] evaluationValue = null;
+        try {
+            coordinateReferenceSystem =
+                    crs != null && !crs.isEmpty() ? CRS.decode(crs) : reader2d.getCoordinateReferenceSystem();
+        } catch (FactoryException e) {
+            throw new WPSException("Could not decode CRS " + crs, e);
+        }
+        Position position = new Position2D(coordinateReferenceSystem, x, y);
+        if (!coordinateReferenceSystem.equals(reader2d.getCoordinateReferenceSystem())) {
+            try {
+                MathTransform transform =
+                        CRS.findMathTransform(coordinateReferenceSystem, reader2d.getCoordinateReferenceSystem(), true);
+                transform.transform(position, position);
+            } catch (FactoryException | TransformException e) {
+                throw new WPSException("Unable to convert select coordinate reference system to grid CRS ", e);
+            }
+        }
+        for (String n : names) {
+            ci = catalog.getCoverageByName(n);
+            if (ci == null) {
+                throw new WPSException("Could not find coverage " + n);
+            }
+            reader = ci.getGridCoverageReader(null, null);
+            reader2d = (GridCoverage2DReader) reader;
+            if (reader2d.getOriginalEnvelope().contains(position)) {
+                try {
+                    gridCoverage2D = getGridCoverage2D(reader2d, position);
+                } catch (TransformException e) {
+                    throw new WPSException("Could not transform point " + position, e);
+                }
+                evaluationValue = convertPrimitiveArrayToNumberArray(gridCoverage2D.evaluate(position));
+                nodata = checkNoData(gridCoverage2D, evaluationValue);
+                if (!nodata) {
+                    break;
+                }
+            }
+        }
+        if (gridCoverage2D == null || nodata) {
+            return new ValuesAtPoint(new Double[] {});
+        }
+
+        return new ValuesAtPoint(evaluationValue);
+    }
+
+    private static GridCoverage2D getGridCoverage2D(GridCoverage2DReader reader, Position position)
```

Comment:
More explicit naming maybe?

```suggestion
    private static GridCoverage2D readAroundPosition(GridCoverage2DReader reader, Position position)
```

Reply:
done